### PR TITLE
FEDX-1472: added build action and fixed publish validate

### PR DESCRIPTION
**Pull request by matthewnitschke-wk**
# [FEDX-1472](https://jira.atl.workiva.net/browse/FEDX-1472)
Issue Status

- Added a `build.yaml` action to facilitate providing an sbom as required by our build system
- Fixed issues with publish validation, where we determine the branch name of the release PR by using the current pubspec.yaml's version and name fields
- Minor cleanup, thanks to yq being installed everywhere by defailt

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .github/workflows/test-unit.yaml [hunk 1]
```diff
@@ -44,10 +44,6 @@ jobs:
       has-chrome-platform: ${{ steps.analyze.outputs.HAS_CHROME_PLATFORM }}
     steps:
       - uses: actions/checkout@v4
-      - name: Install yq
-        run: |
-          sudo wget -O /usr/local/bin/yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64
-          sudo chmod +x /usr/local/bin/yq
       - working-directory: ${{ inputs.package-path }}
```
Comment (matthewnitschke-wk) [inline .github/workflows/test-unit.yaml:47]: github runners have this installed by default https://github.com/actions/runner-images/issues/5144

File: README.md [hunk 2]
```diff
@@ -22,6 +22,10 @@ jobs:
   checks:
     uses: Workiva/gha-dart-oss/.github/workflows/checks.yaml@v1.0.0
 
+  # Generates sbom and uploads it using anchore/sbom-action
+  build:
+    uses: Workiva/gha-dart-oss/.github/workflows/build.yaml@v1.0.0
```
Comment (matthewnitschke-wk) [inline README.md:27]: we might add more to this one day, for now it just uploads the sbom

File: .github/workflows/checks.yaml [hunk 3]
```diff
@@ -61,14 +61,21 @@ jobs:
 
   validate-publish:
     runs-on: ubuntu-latest
-    # only run on release pull requests
-    if: ${{ startsWith(github.ref, 'refs/head/release') && contains(github.event.sender, 'rmconsole') }}
```
Comment (matthewnitschke-wk) [inline .github/workflows/checks.yaml:65]: This wasn't working, opting for a different approach, calculating the release branchname using the current pubspec.yaml

**Comment by matthewnitschke-wk** [comment]
QA +1

- [x]  CI does indeed pass

🚀 @Workiva/release-management-p 🚢

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**